chrome/content/zotero/itemTree.jsx

```diff
@@ -3903,6 +3998,76 @@ var ItemTree = class ItemTree extends LibraryTree {
 			resolve();
 		}));
 	}
+
+	// Check if text quoted in an annotation row contains CJK characters to determine
+	// if it should be italicized or not
+	_containsCJKCharacters = (str) => {
+		let cjkRegexArray = [
+			new RegExp("[\u2E80-\u2EFF]"), // CJK Radicals Supplement
+			new RegExp("[\u3000-\u303F]"), // CJK Symbols and Punctuation
+			new RegExp("[\u3040-\u309F]"), // Hiragana (Japanese)
+			new RegExp("[\u30A0-\u30FF]"), // Katakana (Japanese)
+			new RegExp("[\u1100-\u11FF]"), // Hangul Jamo (Korean)
+			new RegExp("[\u3200-\u32FF]"), // Enclosed CJK Letters and Months
+			new RegExp("[\u3300-\u33FF]"), // CJK Compatibility
+			new RegExp("[\u3400-\u4DBF]"), // CJK Unified Ideographs Extension A
+			new RegExp("[\u4E00-\u9FFF]"), // CJK Unified Ideographs
+			new RegExp("[\uAC00-\uD7AF]"), // Hangul Syllables (Korean)
+			new RegExp("[\uF900-\uFAFF]"), // CJK Compatibility Ideographs
+			new RegExp("[\uFE30-\uFE4F]"), // CJK Compatibility Forms
+			new RegExp("[\u{20000}-\u{2A6DF}]", "u"), // CJK Unified Ideographs Extension B
+			new RegExp("[\u{2F800}-\u{2FA1F}]", "u") // CJK Compatibility Ideographs Supplement
+		];
+		return cjkRegexArray.some(exp => exp.test(str));
```

Comment:
`u` was required on the last two because without it `Range out of order in character class` error is thrown. It didn't seem necessary for other expressions...

And the only reason I did it as an array is for readability. This way, it's easier to see which regex refers to which range of characters, which I think might be helpful in the future if we ever need to go back to edit this to include/exclude some more ranges. With primitive syntax, I don't believe we can break the expression into multiple lines to place comments next to each range, can we? In that case, we would have something like this which to me felt a bit harder to work with:
```
// CJK Radicals Supplement: \u2E80-\u2EFF
// CJK Symbols and Punctuation: \u3000-\u303F
// Hiragana (Japanese): \u3040-\u309F
// Katakana (Japanese): \u30A0-\u30FF
// ... names of remaining ranges 
let cjkRegex = /[\u2E80-\u2EFF\u3000-\u303F\u3040-\u309F\u30A0-\u30FF\u1100-\u11FF\u3200-\u32FF\u3300-\u33FF\u3400-\u4DBF\u4E00-\u9FFF\uAC00-\uD7AF\uF900-\uFAFF\uFE30-\uFE4F\u{20000}-\u{2A6DF}\u{2F800}-\u{2FA1F}]/u;
```

In any case, there was never any need to have multiple `RegExp`. It could be a single `RegExp` with a large concatenated string inside. Would that be better? Or do you think it's worth to make it one long primitive for performance reasons?

Reply:
Oh, sure! The `Script=Han` would not include all the Japanese and Korean ranges but I think `/[\p{Script=Han}|\p{Script=Hiragana}|\p{Script=Katakana}|\p{Script=Hangul}]/u` should do the trick and is, indeed, pretty self explanatory.